### ISSUE TITLE
Cannot list hosts after upgrade to 4.11.3.0

### ISSUE THREAD

**Issue by MartinEmrich** [apache/cloudstack#3506]
##### ISSUE TYPE
 * Bug Report

##### COMPONENT NAME
~~~
API
~~~

##### CLOUDSTACK VERSION
~~~
4.11.3.0
~~~

##### CONFIGURATION
Cloudstack management server (single node), just upgraded von 4.9.2.0 to 4.11.3.0.

##### OS / ENVIRONMENT
CentOS 6

##### SUMMARY
After the upgrade, I can no longer list hosts.

##### STEPS TO REPRODUCE
Click on "Infrastructure", then on "Hosts"
(Or use cloudmonkey, and run "list hosts")

##### EXPECTED RESULTS
I expect to see a list of hosts (XenServers in my case).

##### ACTUAL RESULTS
I see only a field "no data to show".

With cloudmonkey, I get an empty response.

Log file:
```
2019-07-19 17:00:05,674 DEBUG [c.c.a.ApiServlet] (qtp1445157774-109827:ctx-28a7db3d) ===START===  10.14.4.66 -- GET  command=listHosts&type=Routing&zoneid=03058b62-7ec1-40d6-bc3c-77c818bda870&clusterid=d24d9682-7f37-4113-bccf-4ca803cce3b2&page=1&pagesize=20&response=json&_=1563548405631
2019-07-19 17:00:05,676 DEBUG [c.c.a.ApiServer] (qtp1445157774-109827:ctx-28a7db3d ctx-9df7bb4c) CIDRs from which account 'Acct[48ad12e6-68bd-11e4-89ca-5193ece8c729-admin]' is allowed to perform API calls: 0.0.0.0/0,::/0
2019-07-19 17:00:05,679 DEBUG [c.c.a.q.QueryManagerImpl] (qtp1445157774-109827:ctx-28a7db3d ctx-9df7bb4c) >>>Searching for hosts>>>
2019-07-19 17:00:05,682 DEBUG [c.c.a.q.QueryManagerImpl] (qtp1445157774-109827:ctx-28a7db3d ctx-9df7bb4c) >>>Generating Response>>>
2019-07-19 17:00:05,682 DEBUG [c.c.a.ApiServlet] (qtp1445157774-109827:ctx-28a7db3d ctx-9df7bb4c) ===END===  10.14.4.66 -- GET  command=listHosts&type=Routing&zoneid=03058b62-7ec1-40d6-bc3c-77c818bda870&clusterid=d24d9682-7f37-4113-bccf-4ca803cce3b2&page=1&pagesize=20&response=json&_=1563548405631
```

**Comment by MartinEmrich** [comment]
Looks like a database schema issue: the `hosts` table contents is there, but `SELECT * FROM host_view;` gives no result.

**Comment by MartinEmrich** [comment]
After comparing with a working cluster, I found the issue:

The `host_view` contains lots of LEFT JOINs. After removing the joins on last_annotation_view, I got results.
On both servers, this view is empty, but on the other one, I got results.
Then I noticed that the other server runs MariaDB 10.1.38, while the faulty server runs 10.1.28.

So I upgraded MariaDB, and now it works, so it apparently was a bug in MariaDB.
